workshop/content/basics/templates/pseudo-parameters/_index.md

```diff
@@ -0,0 +1,302 @@
+
+![lambda-test](pseudo-parameters/lambda-test.png)
+After you invoke the Lambda function, under Function Logs (or Log output) section you will see some output text like the example below:
+
+![lambda-png](pseudo-parameters/lambda.png)
+
+'alice' is the value of the 'dbUsername' SSM parameter and the Lambda function has logic to print the value of `dbUsername` parameter when invoked. If you see a test output that prints `SSM dbUsername parameter value: alice` like in the picture above, you have verified the SSM parameter policy that you constructed using pseudo parameters is working as you expected.
+
+
+### Challenge
+You’ve now learned how to leverage pseudo parameters in CloudFormation templates. Now let’s say you want to add an S3 bucket to the CloudFormation template. Let’s say you choose the name of the S3 bucket to be as in the following format:  YourPrefix-RegionName-AWSAccount ID. Example `my-demo-bucket-us-east-1-111122223333`
+
+**Task:** Describe an S3 bucket resource in your template, and use a combination of CloudFormation parameters and pseudo parameters to construct the bucket name like in the example above. You want to add a prefix for the bucket name, passed in as a CloudFormation parameter, so that your bucket name will be composed as follows: `${yourprefix}-${AWS::AccountId}-${AWS::Region}`
+
+{{%expand "Need a hint?" %}}
+
+- See the documentation for [CloudFormation parameters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html) and define the parameter for s3 bucket prefix in a similar way.
+- CloudFormation parameters are referenced in a similar way to how we referenced pseudo parameters inside of the `!Sub` function above. For example, if your parameter name is `S3BucketNamePrefix` you can reference this inside of `!Sub` function as follows `!Sub '${S3BucketNamePrefix}'`
+
+
+
+{{% /expand %}}
+
+
+
+{{%expand "Want to see the solution?" %}}
+```YAML
+# add a parameter that will let users input a prefix to be used with s3 buckets
+Parameters:
+  S3BucketNamePrefix:
+    AllowedPattern: ^[0-9a-zA-Z]+([0-9a-zA-Z-]*[0-9a-zA-Z])*$
+    ConstraintDescription: Quick Start bucket name can include numbers, lowercase
+      letters, uppercase letters, and hyphens (-). It cannot start or end with a hyphen
+      (-).
+    Description: The prefix to use for your S3 bucket
+    Type: String
+    MinLength: 3
+    Default: my-demo-bucket
+
+Resources:
+  DemoBucket:
+      Type: 'AWS::S3::Bucket'
+      Properties:
+        BucketName: !Sub '${S3BucketNamePrefix}-${AWS::Region}-${AWS::AccountId}'
+```
+See `code/solutions/pseudo-parameters.yaml` for the full solution.
```

Comment:
You will need to update the path, and include the sub directory where the solution template will be.